app/controllers/simulations_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class SimulationsController < ApplicationController
+  def index
+    @simulations = Simulation.all
+  end
+
+  def show
+    @simulation = Simulation.find(params[:id])
+  end
+
+  def new
+    @simulation = Simulation.new
+  end
+
+  def create
+    @simulation = Simulation.new(simulation_params)
+    if @simulation.save
+      redirect_to @simulation
+    else
+      render :new
+    end
+  end
```

Comment:
This may be an opportunity into setting up some authorized routes in Rails. That is, you can create user logins for authentication, and then authorize only `admin` users to create simulations, for example.  Or allow anyone to create, but only admins can destroy.  Or allow anyone to create, but they can only see the simulations they have created, whereas admins can see all.

This would give you an opportunity to look into [Authorization gems](https://github.com/markets/awesome-ruby#authentication-and-oauth) and [Authentication gems](https://github.com/markets/awesome-ruby#authorization).

For Autentication, [Devise](https://github.com/heartcombo/devise) is very popular, and is built on [warden](https://github.com/hassox/warden). I'd imagine [JWT](https://github.com/jwt/ruby-jwt) is more popular these days, as well, or the [Auth0](https://github.com/lazaronixon/authentication-zero) gem.  I'd recommend Devise since that's what most Rails developers have worked with, probably.

For authorization, I think [Pundit](https://github.com/elabs/pundit) is by far the best and is used in a few places at Shopify. But [CanCanCan](https://github.com/CanCanCommunity/cancancan) is also used at Shopify a bit.

Reply:
Agreed. I was reluctant to go live given the lack of AuthN/Z. But since it is a pet project in a free tier, I just did. 

Thanks for the hints to where to look at. Created #10 to address this.